### ISSUE TITLE
libsvn-hooks-perl: FTBFS: t/02-generic.t: Failed tests:  11-12

### ISSUE THREAD

**Issue by carnil** [gnustavo/SVN-Hooks#3]
Hi,

We have the following bug reported to the Debian package of SVN-Hooks
(https://bugs.debian.org/795756):

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Source: libsvn-hooks-perl
Version: 1.28-1
Severity: serious
Justification: fails to build from source
User: reproducible-builds@lists.alioth.debian.org
Usertags: ftbfs
X-Debbugs-Cc: reproducible-builds@lists.alioth.debian.org

Dear Maintainer,

libsvn-hooks-perl fails to build from source on unstable/amd64:

  [..]
  Test Summary Report
  -------------------
  t/02-generic.t            (Wstat: 512 Tests: 13 Failed: 2)
    Failed tests:  11-12
    Non-zero exit status: 2
  Files=21, Tests=172, 54 wallclock secs ( 0.20 usr  0.06 sys + 35.66
  cusr 11.88 csys = 47.80 CPU)
  Result: FAIL
  Failed 1/21 test programs. 2/172 subtests failed.
  Makefile:947: recipe for target 'test_dynamic' failed
  make[1]: *** [test_dynamic] Error 255
  make[1]: Leaving directory '/tmp/buildd/libsvn-hooks-perl-1.28'
  dh_auto_test: make -j1 test TEST_VERBOSE=1 returned exit code 2
  debian/rules:7: recipe for target 'build' failed
  make: *** [build] Error 2
  dpkg-buildpackage: error: debian/rules build gave error exit status 2

  [..]

The full build log is attached or can be viewed here:

  https://reproducible.debian.net/logs/unstable/amd64/libsvn-hooks-perl_1.28-1.build1.log.gz


Regards,
```
------8<-----------8<-----------8<-----------8<-----------8<-----

The test fails with subversion 1.9.0 installed, but works if I downgrade
to the pervious available version, which was 1.8.13.

Thanks for considering,
  Salvatore Bonaccorso,
  Debian Perl Group

**Comment by gnustavo** [comment]
Thanks, Salvatore.

It seems that this is a bug on Subversion 1.9. I've just sent a message to the users@subversion.apache.org list with the subject "svn (un)lock commands succeed when pre-(un)lock hooks fail on svn 1.9" explaining it in detail.

In summary, the "svn lock" and the "svn unlock" commands succeed on svn 1.9 even if the corresponding hooks (pre-lock and pre-unlock) fail.

I think I'll have to check the subversion version and disregard those two tests if I'm using svn 1.9.x.

**Comment by gnustavo** [comment]
I forgot to mention that the message I sent to the svn list should appear shortly on the [archives](http://mail-archives.apache.org/mod_mbox/subversion-users/201508.mbox/browser).

**Comment by gnustavo** [comment]
Here [it is](http://mail-archives.apache.org/mod_mbox/subversion-users/201508.mbox/%3CCAPx58qoeg20z70tR%2BoJTAEu3ZRPkhRwpesNx%3DUHbPvcqmPosTQ%40mail.gmail.com%3E).

**Comment by gnustavo** [comment]
Hi, Salvatore. I just released SVN::Hooks version 1.29 which should bypass the svn 1.9 bug and have all tests pass.

I'm closing this issue but if you see any problem with this fix, please, comment further.

Thanks !
